internal/controller/pipelinerollout/pipelinerollout_progressive.go

```diff
@@ -108,54 +118,54 @@ func (r *PipelineRolloutReconciler) AssessUpgradingChild(
 	if err != nil {
 		return assessment, reasonFailure, err
 	}
+
 	// if we fail even once, that's considered failure
 	if assessment == apiv1.AssessmentResultFailure {
 		// set AssessmentEndTime to now and return failure
+		numaLogger.Debugf("Assessment failed for upgrading child %s", existingUpgradingChildDef.GetName())
 		_ = progressive.UpdateUpgradingChildStatus(pipelineRollout, func(status *apiv1.UpgradingChildStatus) {
-			assessmentEndTime := metav1.NewTime(time.Now())
-			status.BasicAssessmentEndTime = &assessmentEndTime
+			status.TrialWindowStartTime = nil
+			status.BasicAssessmentEndTime = &metav1.Time{Time: currentTime}
```

Comment:
I'm thinking maybe BasicAssessmentEndTime would be nil here? 

Let's think what `BasicAssessmentStartTime` and `BasicAssessmentEndTime` should be - I'm thinking that say the whole thing was successful - that should probably equal the time of the actual window of success. 

Ideally, if it's unsuccessful, it should represent the time that the window started and then when it failed. 

(Let me know if any of that doesn't sound right.)